File: scripts/verify_model.py
import asyncio

from app.agent import AgentLoop
from app.config import settings
from app.providers.openai_compatible import OpenAICompatibleProvider
from app.ptc import PTCExecutor
from app.tools import create_default_registry


async def main() -> None:
    if settings.is_demo:
        raise SystemExit("未检测到 DASHSCOPE_API_KEY 或 OPENAI_API_KEY。")

    tools = create_default_registry()
    events: list[str] = []

    def record_event(event: dict) -> None:
        label = f"{event['type']}:{event.get('name')}" if event["type"] == "tool_start" else event["type"]
        events.append(label)
        if event["type"] == "tool_end":
            print(f"Tool result ({event.get('name')}): {event.get('result')}")
    agent = AgentLoop(
        provider=OpenAICompatibleProvider(
            api_key=settings.api_key,
            base_url=settings.base_url,
            model=settings.model,
        ),
        tools=tools,
        ptc=PTCExecutor(tools),
        system_prompt="你是连通性检查助手。必须调用工具，并优先使用 execute_ptc。",
        max_steps=8,
    )
    answer = await agent.run(
        [{"role": "user", "content": "用 PTC 获取上海当前时间并计算 6 乘以 7，然后一句话回答。"}],
        record_event,
    )
    print(f"Provider: {settings.provider_name}")
    print(f"Model: {settings.model}")
    print(f"Events: {' -> '.join(events)}")
    print(f"Answer: {answer}")


if __name__ == "__main__":
    asyncio.run(main())

File: scripts/verify-model.js
import { Agent } from "../src/agent/agent.js";
import { OpenAICompatibleProvider } from "../src/agent/provider.js";
import { resolveModelConfig } from "../src/config.js";

const config = resolveModelConfig();
if (!config.apiKey) {
  console.error("未检测到 DASHSCOPE_API_KEY 或 OPENAI_API_KEY。");
  process.exitCode = 1;
} else {
  console.log(`Provider: ${config.providerName}`);
  console.log(`Model: ${config.model}`);

  const agent = new Agent({
    provider: new OpenAICompatibleProvider(config),
    systemPrompt: "你是模型连通性检查助手。必须按用户要求调用可用工具。",
    maxSteps: 4,
  });

  const events = [];
  const answer = await agent.run(
    [{ role: "user", content: "调用工具获取 Asia/Shanghai 当前时间，然后用一句中文回答。" }],
    (event) => events.push(event.type === "tool_start" ? `${event.type}:${event.name}` : event.type),
  );

  console.log(`Events: ${events.join(" -> ")}`);
  console.log(`Answer: ${answer}`);
}
